src/web/components/App.tsx

```diff
@@ -166,6 +168,14 @@ export const App = ({ CAPI, NAV }: Props) => {
         );
     }, []);
 
+    useEffect(() => {
+        setIsRecurringContributor(getCookie('gu_recurring_contributor') === 'true');
+    }, []);
+
+    useEffect(() => {
+        setIsDigitalSubscriber(getCookie('gu_digital_subscriber') === 'true');
+    }, []);
+
     useEffect(() => {
```

Comment:
We're in the process of moving state down from App to child components due to the architecture of partial hydration (essentially setting state causes re-hydration, which is bad...). Could you please move the setState and useEffect into `Links`?

Reply:
OK I've made the change as suggested

Reply:
Fab, thanks!